docs/maui/views/popup/popup-result.md

```diff
@@ -0,0 +1,160 @@
+
+#### Defining your Popup
+
+The easiest way to create a `Popup<T>` is to add a new `.NET MAUI ContentView (XAML)` to your project, this will create 2 files; a _*.xaml_ file and a _*.xaml.cs_ file. The contents of each file can be replaced with the following:
+
+##### .xaml
+
+```xaml
+<toolkit:Popup
+    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+    x:TypeArguments="system:Boolean"
+    x:Class="MyProject.ReturnResultPopup">
+
+    <VerticalStackLayout>
+        <Label Text="This is a very important message! Do you agree?" />
+
+        <Button Text="Yes" 
+                Clicked="OnYesButtonClicked" />
+
+        <Button Text="No"
+                Clicked="OnNoButtonClicked" />
+    
+    </VerticalStackLayout>
+    
+</toolkit:Popup>
+```
+
+##### .xaml.cs
+
+```csharp
+public partial class ReturnResultPopup : ContentView
+{
+    public ReturnResultPopup()
+    {
+        InitializeComponent();
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+}
+```
+
+> [!IMPORTANT]
+> If the code behind file is not created along with the call to `InitializeComponent` then an exception will be thrown when trying to display your `Popup`.
+
+### C#
+
+The following section covers how to create a `Popup` using C#.
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+namespace MyProject;
+
+public class ReturnResultPopup : Popup<bool>
+{
+    public ReturnResultPopup()
+    {
+        var yesButton = new Button { Text = "Yes" };
+        yesButton.Clicked += OnYesButtonClicked;
+
+        var noButton = new Button { Text = "No" };
+        noButton.Clicked += OnNoButtonClicked;
+
+        Content = new VerticalStackLayout
+        {
+            Children = 
+            {
+                new Label { Text = "This is a very important message! Do you agree?" },
+                
+                yesButton,
+                noButton
+            }
+        };
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+```
+
+The `Close` method allows for an `object` value to be supplied, this will be the resulting return value. 
+
+## Awaiting the result from a Popup
+
+In order to await the result the `ShowPopupAsync` method must be used as follows:
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+public class MyPage : ContentPage
+{
+    public async Task DisplayPopup()
+    {
+        var popup = new ReturnResultPopup();
+
+        var result = await this.ShowPopupAsync(popup, CancellationToken.None);
```

Comment:
If I can remember well, the `ShowPopupAsync` doesn't return an `object` it returns an `IPopupResult<T>` instead, so the cast would never be `true`, right?